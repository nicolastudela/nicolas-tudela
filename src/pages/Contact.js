import React, { useEffect } from 'react'
import { TitleAndSubtitle, ContentPanel } from 'components'
import { Box, Typography } from '@smooth-ui/core-em'
import { Space } from 'uiCommons'
import { useClientDeviceType } from 'components/utils/useClientDeviceType'
import { gmailLogo, skypeLogo, backgrKeyword } from 'images'
import linkedinBadgeCreator from '../utils/linkedinBadgeCreator'

const LinkedinBox = () => {
  useEffect(() => {
    linkedinBadgeCreator(window)
  }, [])

  return (
    <div
      className="LI-profile-badge"
      data-version="v1"
      data-size="large"
      data-locale="es_ES"
      data-type="horizontal"
      data-theme="light"
      data-vanity="nicolas-tudela-85455a2b"
    >
      <a
        className="LI-simple-link"
        href="https://ar.linkedin.com/in/nicolas-tudela-85455a2b?trk=profile-badge"
      >
        Nicolas Tudela
      </a>
    </div>
  )
}

const Contact = () => {
  const { isMobile } = useClientDeviceType()
  return (
    <>
      <ContentPanel color="lightBlue">
        {({ color }) => (
          <TitleAndSubtitle title="Conect with me today" color={color} />
        )}
      </ContentPanel>
      <ContentPanel
        display="flex"
        flexDirection="row"
        alignItems="center"
        color="white"
        justifyContent="space-between"
      >
        <LinkedinBox />
        <Box>
          <a
            tabIndex="0"
            display="block"
            target="_blank"
            rel="noopener noreferrer"
            href="mailto:nicolastudela@gmail.com"
          >
            <img
              alt="gmailLogo"
              src={gmailLogo}
              style={{
                height: '256px',
                width: '256px',
              }}
            />
          </a>
        </Box>
        <Box>
          <a
            tabIndex="0"
            display="block"
            target="_blank"
            rel="noopener noreferrer"
            href="https://join.skype.com/invite/BfPs9S9gdahK"
          >
            <img
              alt="skypeLogo"
              src={skypeLogo}
              style={{
                height: '256px',
                width: '256px',
              }}
            />
          </a>
        </Box>
      </ContentPanel>
      <ContentPanel
        color="black"
        backgroundImage={`url('${backgrKeyword}')`}
        backgroundPosition="center"
        width={{ sm: 6 / 7, lg: 3 / 4 }}
        display="flex"
      >
        {!isMobile && (
          <>
            <Space kind="inline" size="xxl" />
            <Space kind="inline" size="xxl" />
          </>
        )}
        <Typography
          css={{ textShadow: '2px 2px 2px rgba(0,153,204,.9)' }}
          color="white"
          fontFamily="'Lobster',cursive"
          fontSize="1.5em"
          as="div"
        >
          Feel free to reach out to me if you have any questions or comments
          about the services I offer. Also, you could always just send me a
          friendly hello.
        </Typography>
      </ContentPanel>
    </>
  )
}

export default Contact
