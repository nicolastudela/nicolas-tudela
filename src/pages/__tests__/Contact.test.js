import React from 'react'
import renderer from 'react-test-renderer'
import BasicTestWrapper from 'test-utils'
import Contact from '../Contact'

describe('contact page', () => {
  it('renders propery', () => {
    expect.assertions(1)
    const component = renderer.create(
      <BasicTestWrapper>
        <Contact />
      </BasicTestWrapper>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
