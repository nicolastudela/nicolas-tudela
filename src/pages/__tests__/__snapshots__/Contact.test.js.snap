// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact page renders propery 1`] = `
Array [
  <div
    className="sui-box css-1nkjupr"
  >
    <div
      className="sui-box css-p74edf"
    >
      <h1
        className="sui-typo css-17fvmi5"
      >
        Conect with me today
      </h1>
    </div>
  </div>,
  <div
    className="sui-box css-2zoooe"
  >
    <div
      className="sui-box css-1oswc43"
    >
      <div
        className="LI-profile-badge"
        data-locale="es_ES"
        data-size="large"
        data-theme="light"
        data-type="horizontal"
        data-vanity="nicolas-tudela-85455a2b"
        data-version="v1"
      >
        <a
          className="LI-simple-link"
          href="https://ar.linkedin.com/in/nicolas-tudela-85455a2b?trk=profile-badge"
        >
          Nicolas Tudela
        </a>
      </div>
      <div
        className="sui-box css-8qkufk"
      >
        <a
          display="block"
          href="mailto:nicolastudela@gmail.com"
          rel="noopener noreferrer"
          tabIndex="0"
          target="_blank"
        >
          <img
            alt="gmailLogo"
            src="test-file-stub"
            style={
              Object {
                "height": "256px",
                "width": "256px",
              }
            }
          />
        </a>
      </div>
      <div
        className="sui-box css-8qkufk"
      >
        <a
          display="block"
          href="https://join.skype.com/invite/BfPs9S9gdahK"
          rel="noopener noreferrer"
          tabIndex="0"
          target="_blank"
        >
          <img
            alt="skypeLogo"
            src="test-file-stub"
            style={
              Object {
                "height": "256px",
                "width": "256px",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>,
  <div
    className="sui-box css-ga31l5"
  >
    <div
      className="sui-box css-z1ly8m"
    >
      <div
        className="sui-box css-1dhq5gn"
      />
      <div
        className="sui-box css-1dhq5gn"
      />
      <div
        className="sui-typo css-1vubwr0-Contact"
      >
        Feel free to reach out to me if you have any questions or comments about the services I offer. Also, you could always just send me a friendly hello.
      </div>
    </div>
  </div>,
]
`;
